internal/commands/launch.go

```diff
@@ -84,8 +85,14 @@ func (l *LaunchCommand) Execute() error {
 
 	logs.Debugf("Filtered environment variables")
 
+	// 4. wait for n8n instance to be ready
+
+	if err := http.WaitForN8n(n8nURI); err != nil {
+		return fmt.Errorf("exhausted retries while waiting for n8n instance to be ready: %w", err)
+	}
```

Comment:
I think we should also use retry whenever we try to fetch a grant token and not only in startup. For example:

1. Task runner has been started
2. n8n crashes
3. Task runner disconnects and exits
4. Launcher tries to connect to n8n, this will fail since n8n is down

We can still in addition keep this here if we want to.

Reply:
As discussed, went with unlimited retries for startup readiness check (to account for long-running migrations) and limited retries for token exchange (to account for crashes and restarts).